stubs/CoreGenericIterators.phpstub

```diff
@@ -355,10 +355,10 @@ class CachingIterator extends IteratorIterator implements OuterIterator , ArrayA
     const TOSTRING_USE_CURRENT = 4 ;
     const TOSTRING_USE_INNER = 8 ;
     const FULL_CACHE = 256 ;
-    
+
     /**
      * @param TIterator $iterator
-     * @param int-mask-of<self::*> $flags
+     * @param int-mask<1,2,4,8,16,256> $flags
```

Comment:
Why change this? The previous type should work the same way but is more flexible to future changes, no?

Reply:
But then we need to provide a template when creating the CachingIterator, and this is what we want to avoid based on what I read on previous comments, no ?

Reply:
unless I'm mistaken, `int-mask-of<self::*>` should not need any template

Reply:
I don't know, could you confirm?

Reply:
Well, that's what it had before this PR, so if it works on psalm.dev right now it should be fine: https://psalm.dev/r/ecacca8230

Reply:
Fixed.